### PR TITLE
Remove unnecessary sleep for 2 seconds in various places in MR.

### PR DESCRIPTION
**Pull request by chtyim**
- The sleep was there for waiting metrics be published. However, on
  stopping the metrics collection service, metrics are being force pushed,
  hence the sleep is unnecessary

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapperWrapper.java [hunk 1]
```diff
@@ -135,8 +132,11 @@ public void run(Context context) throws IOException, InterruptedException {
       }
 
     } finally {
-      basicMapReduceContext.close();
-      basicMapReduceContext.getMetricsCollectionService().stop();
+      try {
+        basicMapReduceContext.close();
+      } finally {
+        basicMapReduceContext.getMetricsCollectionService().stop();
```
Comment (anew) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapperWrapper.java]: so why does the context not stop the metrics collection service?

Comment (anew) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapperWrapper.java]: I guess it is a bigger change... ok

**Comment by anew** [comment]
LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapperWrapper.java (hunk 1 above)
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapperWrapper.java]: It cannot. There are multiple places that call BasicMapReduceContext.close(), and in some context (e.g. MapReduceRuntimeService in local mode), it shares the same instance as the system metrics collection service. The current code base is pretty messy around creation and closing of the MR context object. It's part of the refactoring tracked by https://issues.cask.co/browse/CDAP-2937

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-RBT340-1

**Pull request merged**